internal/controllers/control/cassandratask_controller.go

```diff
@@ -815,6 +814,13 @@ func (r *CassandraTaskReconciler) reconcileEveryPodTask(ctx context.Context, dc
 					<-jobRunner
 				}()
 
+				if taskConfig.SyncFeature != "" {
+					if !features.Supports(taskConfig.SyncFeature) {
+						logger.Error(err, "Pod doesn't support this feature", "Pod", pod, "Feature", taskConfig.SyncFeature)
+						jobStatus.Status = podJobError
+					}
```

Comment:
issue: This would still run the SyncFunc  in line 824 as nothing prevents it we simply log and mark it failed and let the next section cause the failure.

Reply:
Thanks, fixed.